From eaeeba690e227aa97d5ff2cb4b06fd3be4db12a5 Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Tue, 14 May 2024 17:18:13 +0800
Subject: [PATCH 129/206] mtk: mt76: mt7996: Not to remove link when link 0 is
 the default link

This statement is a temporary solution to prevent conflict when
changing between MLD and non-MLD AP in the past. Now we have a
formal solution in mt7996_assign_vif_chanctx(), so the statement
can be deleted.

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 mt7996/main.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/mt7996/main.c b/mt7996/main.c
index b063e455..700a0396 100644
--- a/mt7996/main.c
+++ b/mt7996/main.c
@@ -2618,9 +2618,6 @@ mt7996_change_vif_links(struct ieee80211_hw *hw, struct ieee80211_vif *vif,
 
 	mutex_lock(&dev->mt76.mutex);
 
-	/* check if there's legacy bss needed to be removed */
-	if (rcu_access_pointer(mvif->link[0]) == &mvif->deflink)
-		rem |= BIT(0);
 	/* remove first */
 	for_each_set_bit(link_id, &rem, IEEE80211_MLD_MAX_NUM_LINKS) {
 		struct mt7996_bss_conf *mconf =
-- 
2.18.0

